app/utilities/schema.py

```diff
@@ -177,7 +178,44 @@ def load_schema_from_url(schema_url, language_code):
 
     constructed_schema_url = f"{schema_url}?language={language_code}"
 
-    req = requests.get(constructed_schema_url)
+    session = requests.Session()
+
+    retries = Retry(
+        total=3,
+        allowed_methods=frozenset({"GET"}),
+        backoff_factor=0.1,
+        status_forcelist=[
+            408,
+            429,
+            500,
+            501,
+            502,
+            503,
+            504,
+            505,
+            506,
+            507,
+            509,
+            510,
+            511,
+        ],
+    )  # Codes to retry according to Google Docs
+
+    session.mount("https://", HTTPAdapter(max_retries=retries))
+
+    try:
+        req = session.get(constructed_schema_url, timeout=3)
+    except Exception as exc:
+        raise ConnectionError from exc
```

Comment:
We should avoid catching a generic exception, it is usually bad practice. We should handle the errors that can we expect to occur. The rest should get raised.

All exceptions requests can raise will be caught under `RequestException`. Does the retry object retry ConnectionError implicitly since that doesn't return a status code?. Otherwise, we would need to catch and handle connection error first, as it won't strictly return a status code. Therefore, making the request with retry its own method would be beneficial as you need to manually retry.

Reply:
Changed `Exception` to `RequestException`. I went through some topics about `Retry` and `ConnectionError` indeed  is being implicitly retried despite no status code, so it will work fine here. Thanks!